quizmaster-client-app/src/components/QuizHosting/QuestionMarker.tsx

```diff
@@ -285,22 +297,24 @@ export default function QuestionMarker(props: {
   };
 
   const handleClick = (event: React.MouseEvent<unknown>, name: string) => {
-    const selectedIndex = selected.indexOf(name);
-    let newSelected: string[] = [];
+    const newManuallySelected = new Set(manuallySelected);
+    const newManuallyDeselected = new Set(manuallyDeselected);
 
-    if (selectedIndex === -1) {
-      newSelected = newSelected.concat(selected, name);
-    } else if (selectedIndex === 0) {
-      newSelected = newSelected.concat(selected.slice(1));
-    } else if (selectedIndex === selected.length - 1) {
-      newSelected = newSelected.concat(selected.slice(0, -1));
-    } else if (selectedIndex > 0) {
-      newSelected = newSelected.concat(
-        selected.slice(0, selectedIndex),
-        selected.slice(selectedIndex + 1),
-      );
+    if (newManuallySelected.has(name)) {
```

Comment:
Could put the delete in the if as it returns true/false if the item exists and is deleted or not. Good to see the array manipulation go.